File: examples/access/constraints/elevation_with_max.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Create an elevation constraint with both minimum and maximum elevation limits for side-looking sensors.
"""

import brahe as bh

# Side-looking sensor with elevation range
constraint = bh.ElevationConstraint(min_elevation_deg=10.0, max_elevation_deg=80.0)

print(f"Created: {constraint}")
# Created: ElevationConstraint(10.00° - 80.00°)

File: examples/access/constraints/elevation_with_max.rs
//! Create an elevation constraint with both minimum and maximum elevation limits for side-looking sensors.

#[allow(unused_imports)]
use brahe as bh;

fn main() {
    // Side-looking sensor with elevation range
    let constraint = bh::ElevationConstraint::new(
        Some(10.0),
        Some(80.0)
    ).unwrap();

    println!("Created: {}", constraint);
    // Created: ElevationConstraint(10.00° - 80.00°)
}
